pyjob/sge.py

```diff
@@ -110,8 +127,12 @@ def _create_runscript(self):
             cmd = '-S {}'.format(self.shell)
             runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
         if self.nprocesses:
-            cmd = '-pe {} {}'.format(self.environment, self.nprocesses)
-            runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
+            if self.environment in self.available_environments:
```

Comment:
should this call be made here? how about we validate in the constructor? given that a user requests a given environment,  we shouldn't just silently fail but fail upon incorrect setting immediately.

one possible solution could be to add a `dependencies` property or so to the base class. each inherited `Task` class could override. then we evaluate at constructor level that all dependencies are satisfied.

WDYT?

Reply:
I think this is a good idea. I'm not sure what the dependencies for the other `ClusterTask` would be since I only have access to SGE and SLURM, but it's something that I could add.

Reply:
Latest commit should solve this, the only requirement I'm checking on those systems I have no experience is if a basic queue command is available.